lib/app_config.ex

```diff
@@ -132,21 +133,41 @@ defmodule AppConfig do
   ## Example
 
       {:ok, "VALUE"} = #{inspect __MODULE__}.fetch_env(:my_app, :test_var)
+      :ok = Application.put_env(:my_test_app, :test_var_1, %{ test_var_2: "VALUE" })
+      {:ok, "VALUE"} = #{inspect __MODULE__}.fetch_env(:my_app, [ :test_var_1, :test_var_2 ])
 
   """
-  @spec fetch_env(app | Keyword.t, key) :: {:ok, value} | :error
-  def fetch_env(env, key) when (is_atom(env) or is_list(env)) and is_atom(key) do
+  @spec fetch_env(app | Keyword.t, key | List.t) :: {:ok, value} | :error
+  def fetch_env(env, key) when (is_atom(env) or is_list(env) or is_map(env)) and is_atom(key) do
     with {:ok, value} <- fetch(env, key) do
       get_env_value(value)
     end
   end
+  def fetch_env(env, [ first_key | keys ]) when is_atom(env) or is_list(env) or is_map(env) do
```

Comment:
@DrunkenInfant Could you keep the spacing in brackets and braces consistent with the rest of the code? In this case, the expression should be `[first_key | keys]`.